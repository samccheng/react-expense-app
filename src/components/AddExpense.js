import React, { Component } from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { addStartExpense } from '../actions/expenses'

export class AddExpense extends Component {
  onSubmit = (expense) => {
    this.props.addStartExpense(expense)
    this.props.history.push('/dashboard')
  }

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">add expense</h1>
          </div>
        </div>

        <div className="content-container">
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    )
  }
}


const mapDispatchToProps = (dispatch) => ({
  addStartExpense: (expense) => dispatch(addStartExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(AddExpense)
