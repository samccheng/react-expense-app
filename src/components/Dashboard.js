import React, { Component } from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

export default class Dashboard extends Component {
  render() {
    return (
      <div>
        <ExpenseListFilters />
        <ExpenseList />
      </div>

    )
  }
}
