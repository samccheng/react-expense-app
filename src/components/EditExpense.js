import React, { Component } from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

export class EditExpense extends Component{
  onSubmit= (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/dashboard')
  }

  onClick = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id })
    this.props.history.push('/dashboard')
  }

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>

        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />

          <button className="button button--edit" onClick={this.onClick}>remove expense</button>
        </div>
      </div>
    )
  }
}


const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense)
