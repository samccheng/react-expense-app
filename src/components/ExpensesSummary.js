import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'


export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">viewing <span>{expensesCount}</span> {expenseWord} totaling <span>{numeral(expensesTotal/100).format('$0,0.00')}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">add expense</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)

  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
