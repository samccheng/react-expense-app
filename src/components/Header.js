import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true} >home</NavLink>
    <NavLink to="/create" activeClassName="is-active" >create expense</NavLink>
    <NavLink to="/help" activeClassName="is-active" >help</NavLink>
  </header>
)

export default Header
