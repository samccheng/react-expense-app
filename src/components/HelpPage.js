import React from 'react'
import { Link } from 'react-router-dom'

const HelpPage = () => (
  <div>
    <p>Help page</p>
    <Link to="/">home</Link>
  </div>
)

export default HelpPage
