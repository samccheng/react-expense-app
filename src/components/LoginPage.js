import React from 'react'

export const LoginPage = () => (
  <div>
    <button>login</button>
  </div>
)
