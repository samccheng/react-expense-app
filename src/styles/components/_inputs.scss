.text-input {
  border: 1px solid #cacccd;
  font-size: $font-size-lg;
  font-weight: 300;
  height: 50px;
  padding: $s-size;
}

.select {
  @extend .text-input;

}

.textarea {
  @extend .text-input;
  height: 10rem;
}
