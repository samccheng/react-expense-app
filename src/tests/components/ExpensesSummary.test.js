import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should show expense summary component with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={100}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should show expense summary component with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={5} expensesTotal={100}/>)
  expect(wrapper).toMatchSnapshot()
})
