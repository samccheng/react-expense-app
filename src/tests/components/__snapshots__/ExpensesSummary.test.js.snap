// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show expense summary component with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $1.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        add expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should show expense summary component with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      viewing 
      <span>
        5
      </span>
       
      expenses
       totaling 
      <span>
        $1.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        add expense
      </Link>
    </div>
  </div>
</div>
`;
